### PR TITLE
Add CREATE_RUN operation to port github action

### PR DESCRIPTION
**Pull request by dvarchev**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ inputs:
     description: 'Client secret'
   operation:
     required: true
-    description: 'One of the following operations: GET, UPSERT, SEARCH, BULK_UPSERT, PATCH_RUN, DELETE'
+    description: 'One of the following operations: GET, UPSERT, SEARCH, BULK_UPSERT, CREATE_RUN, PATCH_RUN, DELETE'
```
Comment (talsabagport) [inline action.yml:17]: Please add `action` to the inputs here

File: src/operations/CreateRunOperation/CreateRunOperation.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import * as core from '@actions/core';
+import assert from 'assert';
+
+import clients from '../../clients';
+import { IOperation } from '../../interfaces';
+import { ActionInput, RunToCreate } from '../../types';
+
+export default class CreateActionOperation implements IOperation {
```
Comment (talsabagport) [inline src/operations/CreateRunOperation/CreateRunOperation.ts:8]:
```suggestion
export default class CreateRunOperation implements IOperation {
```

File: src/operations/OperationFactory/OperationFactory.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { OPERATION_IS_NOT_SUPPORTED } from '../../consts';
 import { IOperation } from '../../interfaces';
 import { ActionInput, OperationType } from '../../types';
+import CreateActionOperation from '../CreateRunOperation/CreateRunOperation';
```
Comment (talsabagport) [inline src/operations/OperationFactory/OperationFactory.ts:4]:
```suggestion
import CreateRunOperation from '../CreateRunOperation/CreateRunOperation';
```

File: src/operations/OperationFactory/OperationFactory.ts [hunk 4]
```diff
@@ -19,6 +20,8 @@ export default class OperationFactory {
 				return new EntitiesSearchOperation(input);
 			case OperationType.BulkUpsert:
 				return new EntityBulkUpserterOperation(input);
+			case OperationType.CreateRun:
+				return new CreateActionOperation(input);
```
Comment (talsabagport) [inline src/operations/OperationFactory/OperationFactory.ts:24]:
```suggestion
				return new CreateRunOperation(input);
```

**Review by talsabagport — CHANGES_REQUESTED** [review]
Looks good overall! 👍  Left few comments

File: action.yml (hunk 1 above)
Comment (dvarchev) [reply action.yml:17]: Added the action input.

File: src/operations/CreateRunOperation/CreateRunOperation.ts (hunk 2 above)
Comment (dvarchev) [reply src/operations/CreateRunOperation/CreateRunOperation.ts]: Fixed the naming as requested

File: src/operations/OperationFactory/OperationFactory.ts (hunk 4 above)
Comment (dvarchev) [reply src/operations/OperationFactory/OperationFactory.ts]: Fixed the naming as requested

File: src/operations/OperationFactory/OperationFactory.ts (hunk 3 above)
Comment (dvarchev) [reply src/operations/OperationFactory/OperationFactory.ts]: Fixed the naming as requested

**Comment by dvarchev** [comment]
Hey @talsabagport,
Thanks a lot for the quick review and comments. I addressed/fixed all of them. Please have a second look whenever you can.

**Comment by dvarchev** [comment]
The tests fail due to missing ClientID. Looking at the workflow it appears that the secrets configuration is incorrect / missing.

File: action.yml [hunk 5]
```diff
@@ -62,6 +62,9 @@ inputs:
   logMessage:
     required: false
     description: 'The port action log message'
+  action:
+    required: false
+    description: 'The port action identifier to creare run for'
```
Comment (talsabagport) [inline action.yml:67]:
```suggestion
    description: 'The port action identifier to create run for'
```

**Review by talsabagport — COMMENTED** [review]
The tests are failing because the secrets are not available to your forked repository, have you tested it at least locally?

**Comment by dvarchev** [comment]
Yes, 
I have tested the PR locally. Apart from the added tests, I also tested the GA in a separate workflow using these steps:
```
      - name: Trigger test action for entity
        uses: officernd/port-github-action@main
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}
          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
          operation: CREATE_RUN
          icon: GithubActions
          blueprint: "gh-action-test-bp"
          action: "gh-action-test-entity"
          identifier: "gh-action-test-bp-entity"
          properties: |-
            {
              "text": "test", 
              "number": 1, 
              "boolean": true
            }
      - name: Trigger test create action
        uses: officernd/port-github-action@main
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}
          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
          operation: CREATE_RUN
          icon: GithubActions
          blueprint: "gh-action-test-bp"
          action: "gh-action-test"
          properties: |-
            {
              "text": "test", 
              "number": 1, 
              "boolean": true
            }
```

Unfortunately not all automated tests pass when executed on my side. Some of those tests rely on specific runIds (like `runId: 'r_1HDz2pBPYltPcfA1`) that I cannot setup on my port account so that they pass with my test clientID.

**Review by talsabagport — APPROVED** [review]

**Pull request merged**